src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState<User | null>();
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const handleOnClose = () => {
+    dispatch(currentTodoSlice.actions.removeTodo(null));
+    setUser(null);
+  };
+
+  useEffect(() => {
+    getUser(selectedTodo.userId)
+      .then(data => setUser(data))
```

Comment:
```suggestion
      .then(setUser)
```

Reply:
Idk why the tests fail, when i do it manually everything's fine